### PR TITLE
[Merged by Bors] - feat: various linear algebra results

### PR DESCRIPTION
**Pull request by ocfnash**
---

I have split these out from https://github.com/leanprover-community/mathlib4/pull/8583 to aid its review. I believe these changes make sense in their own right.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/DirectSum/LinearMap.lean [hunk 1]
```diff
@@ -4,6 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Oliver Nash
 -/
 import Mathlib.Algebra.DirectSum.Module
+import Mathlib.LinearAlgebra.Eigenspace.Basic
```
Comment (ocfnash) [inline Mathlib/Algebra/DirectSum/LinearMap.lean:7]: I note that the only file (currently) importing this one is `Algebra.Lie.Killing` which already depends on `LinearAlgebra.Eigenspace.Basic` (transitively).

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/LinearAlgebra/TensorProduct/Tower.lean [hunk 2]
```diff
@@ -450,3 +450,76 @@ end CommSemiring
...
+
+open TensorProduct
+
+variable {R M₁ M₂ M₃ : Type*} (A : Type*) [CommSemiring R] [Semiring A] [Algebra R A]
+  [AddCommMonoid M₁] [Module R M₁] [AddCommMonoid M₂] [Module R M₂] [AddCommMonoid M₃] [Module R M₃]
+  (f : M₁ →ₗ[R] M₂) (g : M₂ →ₗ[R] M₃)
+
+/-- If `A` is an `R`-algebra, any `R`-linear map `M₁ → M₂` naturally induces an `A`-linear map
+`A ⊗ M₁ → A ⊗ M₂`. -/
+def extendScalars : A ⊗[R] M₁ →ₗ[A] A ⊗[R] M₂ :=
```
Comment (ocfnash) [inline Mathlib/LinearAlgebra/TensorProduct/Tower.lean:464]: To my shame I have just discovered that this duplicates [LinearMap.baseChange](https://github.com/leanprover-community/mathlib4/blob/244c4fda94ef8e27711f604be5b3a20ae5786731/Mathlib/RingTheory/TensorProduct.lean#L68).

I will follow up with a PR to remove this definition (and probably expand the comment on `LinearMap.baseChange` to include the words "extension of scalars").

Comment (ocfnash) [reply Mathlib/LinearAlgebra/TensorProduct/Tower.lean:464]: WIP here #8617

**Pull request closed without merging**